libbeat/generator/fields/fields.go

```diff
@@ -0,0 +1,149 @@
+package fields
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+var (
+	generatedFieldsYml = filepath.Join("_meta", "fields.generated.yml")
+)
+
+// YmlFile holds the info on files and how to write them into the global fields.yml
+type YmlFile struct {
+	Path   string
+	Indent int
+}
+
+func collectBeatFiles(beatPath string, fieldFiles []*YmlFile) ([]*YmlFile, error) {
+	commonFields := filepath.Join(beatPath, "_meta", "fields.common.yml")
+	_, err := os.Stat(commonFields)
+	if os.IsNotExist(err) {
+		return fieldFiles, nil
+	} else if err != nil {
+		return nil, err
+	}
+
+	files := []*YmlFile{
+		&YmlFile{
+			Path:   commonFields,
+			Indent: 0,
+		},
+	}
+
+	return append(files, fieldFiles...), nil
+}
+
+func writeGeneratedFieldsYml(beatsPath string, fieldFiles []*YmlFile) error {
+	outPath := path.Join(beatsPath, generatedFieldsYml)
+	f, err := os.Create(outPath)
+	if err != nil {
+		return err
+	}
+	defer f.Close()
+
+	for _, p := range fieldFiles {
+		content, err := ioutil.ReadFile(p.Path)
+		if err != nil {
+			return err
+		}
+
+		content = indent(content, p.Indent)
```

Comment:
This `indent` method looks a bit complex to me, also on most of the operations done to the `content` we are going to be copying in memory all the files several times.

I wonder if we cannot just read line by line and write them with the proper indentation.